test/distributed/assembly_helpers_kernels.cpp

```diff
@@ -0,0 +1,197 @@
+            gko::array<global_index_type> d_original_positions{exec,
+                                                               num_entries};
+
+            gko::kernels::reference::assembly_helpers::count_non_owning_entries(
+                ref, input, row_partition.get(), part, send_count,
+                send_positions, original_positions);
+            gko::kernels::GKO_DEVICE_NAMESPACE::assembly_helpers::
+                count_non_owning_entries(exec, d_input, d_row_partition.get(),
+                                         part, d_send_count, d_send_positions,
+                                         d_original_positions);
+
+            gko::array<global_index_type> send_offsets{ref, num_parts + 1};
+            std::partial_sum(send_count.get_data(),
+                             send_count.get_data() + num_parts,
+                             send_offsets.get_data() + 1);
+            send_offsets.get_data()[0] = 0;
+            gko::array<global_index_type> d_send_offsets{exec, send_offsets};
+            gko::size_type num_send_entries =
+                send_offsets.get_data()[num_parts];
+            gko::array<global_index_type> send_row_idxs{ref, num_send_entries};
+            gko::array<global_index_type> send_col_idxs{ref, num_send_entries};
+            gko::array<value_type> send_values{ref, num_send_entries};
+            gko::array<global_index_type> d_send_row_idxs{exec,
+                                                          num_send_entries};
+            gko::array<global_index_type> d_send_col_idxs{exec,
+                                                          num_send_entries};
+            gko::array<value_type> d_send_values{exec, num_send_entries};
+
+            gko::kernels::reference::assembly_helpers::fill_send_buffers(
+                ref, input, row_partition.get(), part, send_positions,
+                original_positions, send_row_idxs, send_col_idxs, send_values);
+            gko::kernels::GKO_DEVICE_NAMESPACE::assembly_helpers::
+                fill_send_buffers(exec, d_input, d_row_partition.get(), part,
+                                  d_send_positions, d_original_positions,
+                                  d_send_row_idxs, d_send_col_idxs,
+                                  d_send_values);
+
+            GKO_ASSERT_ARRAY_EQ(send_positions, d_send_positions);
+            GKO_ASSERT_ARRAY_EQ(original_positions, d_original_positions);
+            GKO_ASSERT_ARRAY_EQ(send_count, d_send_count);
+            GKO_ASSERT_ARRAY_EQ(send_row_idxs, d_send_row_idxs);
+            GKO_ASSERT_ARRAY_EQ(send_col_idxs, d_send_col_idxs);
+            GKO_ASSERT_ARRAY_EQ(send_values, d_send_values);
+        }
+    }
+
+    std::default_random_engine engine;
+};
+
+TYPED_TEST_SUITE(AssemblyHelpers, gko::test::ValueLocalGlobalIndexTypes,
+                 TupleTypenameNameGenerator);
+
+
+TYPED_TEST(AssemblyHelpers, AddNonLocalEntriesDiagOffdiagEmptyIsSameAsRef)
```

Comment:
```suggestion
TYPED_TEST(AssemblyHelpers, AddNonLocalEntriesEmptyIsSameAsRef)
```
it tests an empty matrix, right? or is it other meaning I miss?